### DISCUSSION TITLE
How to use double pointer in V?

### DISCUSSION THREAD

**Discussion by Anivie** [vlang/v#17640] [Questions and Answers]
Code like this:
```
pub type JNIEnv = &JNINativeInterface_

pub struct JNINativeInterface_ {
	get_version fn (&JNIEnv) int
}

fn miao(env &JNIEnv) {
	println('hello cat from V on ${(*env).get_version(env)}')
}
```

The right way in C is : `(*env)->get_version `,but now V will use `(*env).get_version ` instead.
If you use:`(*(*env)).get_version(env)`, compiler will throw an error:
```
error: invalid indirect of `JNIEnv`, the type `JNINativeInterface_` is not a pointer
```

How can I use this pointer, just like `(*env)->get_version` in C?

**Comment by spytheman** [comment]
try: `pub type JNIEnv = &&JNINativeInterface_`

**Reply by Anivie** [reply]
> try: `pub type JNIEnv = &&JNINativeInterface_`

It will use `env.GetVersion(env)` dirctry, and error will be `error: request for member 'GetVersion' in something not a structure or union`
